src/manifest.rs

```diff
@@ -1,21 +1,97 @@
-pub fn get_package_name(manifest_path: &std::path::Path) -> anyhow::Result<String> {
-    let manifest = std::fs::read_to_string(manifest_path).map_err(|e| {
-        anyhow::format_err!("Failed when reading {}: {}", manifest_path.display(), e)
-    })?;
-    let manifest: toml_edit::Document = manifest
-        .parse()
-        .map_err(|e| anyhow::format_err!("Failed to parse {}: {}", manifest_path.display(), e))?;
-    let package = manifest.get("package").ok_or_else(|| {
```

Comment:
@epage here's the source for the lines you commented on. I'm not that familiar with the `cargo_metadata` crate and I assumed you had looked into it originally so I didn't dig in further. Happy to take a second look.